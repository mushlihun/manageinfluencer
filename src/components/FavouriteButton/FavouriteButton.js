import React from 'react'
import { ReactComponent as HeartIcon } from '../../icons/heart.svg'
import styled from 'styled-components/macro'

export default function FavouriteButton({ onClick, isFavourite }) {
  return (
    <section>
      <ButtonStyled id="like-button" onClick={onClick}>
        <HeartStyled
          fill={isFavourite ? 'var(--darkpink)' : 'var(--lightpink)'}
        />
      </ButtonStyled>
    </section>
  )
}

const ButtonStyled = styled.button`
  background: var(--lightpink);
  border-radius: 50%;
  text-align: center;
  height: 60px;
  width: 60px;
  display: block;
  margin: auto;
  margin-top: 15px;
  border: none;
  stroke: #e693b5;
`

const HeartStyled = styled(HeartIcon)`
  margin-top: 3px;
`
