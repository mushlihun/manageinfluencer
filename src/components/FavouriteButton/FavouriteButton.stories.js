import React from 'react'
import FavouriteButton from './FavouriteButton'
import { ReactComponent as Heart } from '../../icons/heart.svg'
import styled from 'styled-components'

export default {
  title: 'Components/Button',
  component: FavouriteButton,
  argTypes: {
    headline: { control: 'text' },
  },
}

const Template = (args) => <FavouriteButton {...args} />

export const notLiked = Template.bind({})
notLiked.args = {}

export const Liked = () => <FavouriteButtonLike />

const FavouriteButtonLike = styled(Heart)`
  background: var(--lightpink);
  fill: var(--darkpink);
  border-radius: 50%;
  text-align: center;
  height: 60px;
  width: 60px;
  display: block;
  padding: 10px;
  margin: auto;
  margin-top: 15px;
  border: none;
`
