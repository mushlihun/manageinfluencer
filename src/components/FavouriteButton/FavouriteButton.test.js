import React from 'react'
import renderer from 'react-test-renderer'
import FavouriteButton from './FavouriteButton'
import '@testing-library/jest-dom/extend-expect'

describe('FavouriteButton', () => {
  it('renders FavouriteButton correctly', () => {
    const tree = renderer.create(<FavouriteButton />)
    expect(tree).toMatchSnapshot()
  })
})
