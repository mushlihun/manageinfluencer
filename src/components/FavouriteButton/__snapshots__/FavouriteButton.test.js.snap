// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteButton renders FavouriteButton correctly 1`] = `
<section>
  <button
    className="FavouriteButton__ButtonStyled-sc-1d9fn40-0 eBMGRb"
    id="like-button"
  >
    <svg
      className="FavouriteButton__HeartStyled-sc-1d9fn40-1 ejAJbr"
      fill="var(--lightpink)"
    >
      heart.svg
    </svg>
  </button>
</section>
`;
